numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1558,6 +1570,35 @@ pack_inner(const char *inptr,
             inptr += 16;
         }
     }
+#elif defined __ARM_NEON || defined __ARM_NEON__
+    if (in_stride == 1 && element_size == 1 && n_out > 2) {
+        /* don't handle non-full 8-byte remainder */
+        npy_intp vn_out = n_out - (remain ? 1 : 0);
+        vn_out -= (vn_out & 1);
+        for (index = 0; index < vn_out; index += 2) {
+            unsigned int r;
+            npy_uint64 a = *(npy_uint64*)inptr;
```

Comment:
This unsafe cast and may cause BUS error on armhf, since the pointer must be aligned to the size of the data type.

Reply:
I suggest adding a new type to notify the compiler about the alignment status
```C
typedef npy_uint64 uint64_unaligned __attribute((__aligned__(1)));
````
then you can use it for dereferencing
```C
npy_uint64 a = *((uint64_unaligned*)inptr);
````

Reply:
modified.